### PR TITLE
std Error impl for MmapError and volatile_memory Error

### LINKED ISSUE #18: volatile_memory::Error doesn't impl std::error::Error
Using `?` on `VolatileMemory::get_slice` (applied to a `MmapRegion`), in a function returning a generic Error, doesn't work, because `VolatileMemory::get_slice` returns a Result with an error type of `volatile_memory::Error`, which doesn't impl `std::error::Error`.

Fixing this would make it easier to avoid using `.unwrap()`.

### PR DESCRIPTION
**Pull request by fkuchar**
closes #18 and #17

### CODE REVIEW

File: src/mmap.rs [hunk 1]
```diff
@@ -63,6 +65,23 @@ pub enum Error {
     MemoryRegionOverlap,
 }
 
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Error::InvalidGuestRegion => {
+                write!(f, "Adding the guest base address to the length of the underlying mapping resulted in an overflow.")
```
Comment (bonzini) [inline src/mmap.rs]: Please split this long line to maximum 100 characters per line.

File: src/mmap_unix.rs [hunk 2]
```diff
@@ -46,6 +48,29 @@ pub enum Error {
...
+            ),
+            Error::MapFixed => write!(f, "The forbidden `MAP_FIXED` flag was specified"),
+            Error::MappingOverlap => write!(
+                f,
+                "Mappings using the same fd overlap in terms of file offset and length"
+            ),
+            Error::MappingPastEof => write!(
+                f,
+                "The specified file offset and lenght is greater then file lenght"
+            ),
```
Comment (bonzini) [inline src/mmap_unix.rs]: Length, not length.

Comment (fkuchar) [reply src/mmap_unix.rs]: Done. Thanks.

**Comment by bonzini** [comment]
Also please squash the commits into one.

**Comment by fkuchar** [comment]
Should be alright now. Thanks.

File: src/mmap.rs [hunk 3]
```diff
@@ -68,9 +68,11 @@ pub enum Error {
 impl fmt::Display for Error {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match self {
-            Error::InvalidGuestRegion => {
-                write!(f, "Adding the guest base address to the length of the underlying mapping resulted in an overflow.")
-            }
+            Error::InvalidGuestRegion => write!(
+                f,
+                "Adding the guest base address to the length of the underlying mapping
```
[the remark this answers is not in the document]
Comment (fkuchar) [reply src/mmap.rs]: Silly me, sorry about that. Done.

[the remark this answers is not in the document]
Comment (jiangliu) [reply src/mmap.rs]: Please also use 'cargo fmt' to format your changes:)
Otherwise it breaks the CI tests.

**Pull request merged**